### ISSUE TITLE
downloads.download saveAs does not accept a dataUrl

### ISSUE THREAD

**Issue by Solaner** [mdn/webextensions-examples#202]
This is a small WebExtension where I'm trying to save a data URL on the Computer, using the downloads API: [downloads-save-dataurl.zip](https://github.com/mdn/webextensions-examples/files/887502/downloads-save-dataurl.zip)

It works as expected in the Google Chrome Browser but it doesn't work in Firefox.
In Firefox I'm getting the exception:
"Type error for parameter options (Error processing url: Error: Access denied for URL ...".

I have found Bugzilla entries about similar problems, but they have the status _resolved_.
Could somebody give me a hint, if this is a different error on Firefox and if there is a workaround to have the problem solved (Perhaps Firefox needs different permissions in the Manifest?).

**Comment by andymckay** [comment]
You'll want to file a bugzilla bug and reference the other bugzilla bugs, this repository and its issues are used for the examples contained within. If I understand the problem passing a blob through to downloads should work.

**Comment by Solaner** [comment]
Thank you for your response and sorry for using the wrong forum.
I've been starting with the [latest-download](https://github.com/mdn/webextensions-examples/tree/master/latest-download) example.
After doing the changes for my desired functionality I have not realized that the code was already to far away from the example.
However, with your hint, using a blob, I got a workaround, which still is throwing an exception, but actually is working.
I have filed [Bug 1353546](https://bugzilla.mozilla.org/show_bug.cgi?id=1353546).
